t/42-screenshots.t

```diff
@@ -66,9 +69,14 @@ is_deeply(
     'screenshot not directly cleaned up after deleting job'
 ) or diag explain \@screenshot_data;
 
-# limit job results (which involves deleting unused screenshots)
+# limit screenshots
+my %args = (
+    min_screenshot_id     => $screenshots->search(undef, {rows => 1, order_by => {-asc  => 'id'}})->first->id,
+    max_screenshot_id     => $screenshots->search(undef, {rows => 1, order_by => {-desc => 'id'}})->first->id,
+    screenshots_per_batch => OpenQA::Task::Job::Limit::SCREENSHOTS_PER_BATCH,
+);
 combined_like(
-    sub { OpenQA::Task::Job::Limit::_limit($t->app); },
+    sub { OpenQA::Task::Job::Limit::_limit_screenshots($t->app, undef, \%args); },
```

Comment:
There are much better ways to run a job in tests than to call a private method.

Reply:
You might even be able to use `run_gru_job` from our test utils. https://github.com/os-autoinst/openQA/blob/324e1125073b62a38d45cd50679309e599c767bc/t/lib/OpenQA/Test/Utils.pm#L461